generated/rflx-arrays-generic_inner_message.ads

```diff
@@ -216,14 +216,18 @@ is
 
    generic
       with procedure Process_Payload (Payload : out Types.Bytes);
+      with function Check_Length_Payload (Length : Types.Length) return Boolean;
    procedure Set_Payload (Ctx : in out Context) with
      Pre =>
        not Ctx'Constrained
        and then Has_Buffer (Ctx)
        and then Valid_Next (Ctx, F_Payload)
        and then Field_Last (Ctx, F_Payload) <= Types.Bit_Index'Last / 2
        and then Field_Condition (Ctx, (Fld => F_Payload))
-       and then Available_Space (Ctx, F_Payload) >= Field_Length (Ctx, F_Payload),
+       and then Available_Space (Ctx, F_Payload) >= Field_Length (Ctx, F_Payload)
+       and then Field_First (Ctx, F_Payload) mod 8 = 1
+       and then Field_Length (Ctx, F_Payload) mod 8 = 0
+       and then Check_Length_Payload (Types.Length (Field_Length (Ctx, F_Payload) / 8)),
```

Comment:
I would prefer to use `Byte'Size` here instead of hardcoding `8` (although it's probably very unlikely that we will allow bytes of an other size in the future).